recipes/mp-units/0.7.0/conanfile.py

```diff
@@ -0,0 +1,84 @@
+from conans import ConanFile, CMake, tools
+from conans.tools import Version, check_min_cppstd
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+class MPUnitsConan(ConanFile):
+    name = "mp-units"
+    homepage = "https://github.com/mpusz/units"
+    description = "Physical Units library for C++"
+    topics = ("units", "dimensions", "quantities", "dimensional-analysis", "physical-quantities", "physical-units", "system-of-units", "cpp23", "cpp20", "library", "quantity-manipulation")
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake_find_package_multi"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        compiler = self.settings.compiler
+        self.requires("fmt/7.1.3")
+        self.requires("gsl-lite/0.38.0")
+        if compiler == "clang" and compiler.libcxx == "libc++":
```

Comment:
Isn't there a minimum versions for which the libc++ standard library has ranges?

Reply:
Not in llvm-12. Maybe llvm-13 will have a working implementation. Until then nasty hacks are needed:

https://github.com/mpusz/units/blob/43825a531fa91501277161fefd94c50a3cf0ea96/src/core/include/units/bits/external/hacks.h#L69-L214